client/lib/signup/private-by-default.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import debugFactory from 'debug';
+
+const debug = debugFactory( 'calypso:signup:private-by-default' );
+
+interface PrivateByDefaultSiteSettings {
+	flowName?: string;
+}
+
+/**
+ * Should the site be private by default
+ * @param siteSettings PrivateByDefaultSiteSettings
+ * @returns `true` for private by default & `false` for not
+ */
+export function shouldBePrivateByDefault( {
+	flowName = '',
+}: Readonly<PrivateByDefaultSiteSettings> ): boolean {
```

Comment:
_Observation_

Because we destructure the argument, I believe this usage of `Readonly` is redundant:

```suggestion
}: PrivateByDefaultSiteSettings ): boolean {
```

For example:

```ts
export function shouldBePrivateByDefault( { flowName = '' }: Readonly<PrivateByDefaultSiteSettings> ): boolean {
  flowName = flowName + 'all'; // No error, we're not touching a read-only property.
}
// Using the object, `Readonly` may be important:
export function shouldBePrivateByDefault( s: Readonly<PrivateByDefaultSiteSettings> ): boolean {
  s.flowName = s.flowName + 'all'; // Error: Cannot assign to 'flowName' because it is a read-only property.
}
```

Reply:
I think the idea was that folks unfamiliar with the code would be able to easily modify `shouldBePrivateByDefault()` without accidentally breaking things if they were to mutate the settings object. Which I _think_ is why there's the premptive clone that you observed in https://github.com/Automattic/wp-calypso/pull/34888#discussion_r308665878.

I originally suggested using `Readonly` as a way of confidently removing the object cloning.

For me, `Readonly` type is more about communicating the intended interface for the function, so even if there's no mutation happening in the body now, if that's something you want to communicate with future devs then I think it's useful. (I tend to think of inline destrucuring as part of the function body, not the interface, so maybe that's where my thinking is mixed up).

```typescript
export function shouldBePrivateByDefault( settings: Readonly<PrivateByDefaultSiteSettings> ): boolean {
  // "Manual" destructuring but no mutation, is Readonly redundant in this situation too?
  const flowName = settings.flowName;
}
```

In saying that: I usually don't want functions mutating their parameters, but that doesn't mean I want to pepper `Readonly` everywhere throughout the code! So it's a matter of taste and whether it's communicating anything useful.

Reply:
Sounds good 👍